components/prospects/ProspectNotes.tsx

```diff
@@ -0,0 +1,225 @@
+  const [noteToDelete, setNoteToDelete] = useState(null)
+
+  useEffect(() => {
+    let isMounted = true
+    fetch(`/api/prospect/prospect-note?prospectId=${prospectId}`)
+      .then((response) => response.json())
+      .then((data) => {
+        if (isMounted) {
+          setNotes(data)
+        }
+      })
+      .catch((error) => console.error('Error:', error))
+      return () => {isMounted = false}
+  }, [notes])
+
+  function deleteNote(noteId) {
+    fetch('/api/prospect/prospect-note', {
+      method: 'DELETE',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ id: noteId })
+    })
+    .then(() => setNotes(notes.filter((note) => note.id !== noteId)))
+    .catch((error) => console.error('Error:', error))
+  }
+
+  function updateNote(noteId, newNote) {
+    fetch('/api/prospect/prospect-note', {
+      method: 'PUT',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ id: noteId, note: newNote })
+    })
+    .then(() => {
+      const updatedNoteIndex = notes.findIndex((note) => note.id === noteId)
+      const updatedNote = { ...notes[updatedNoteIndex], note: newNote }
+      const updatedNotes = [...notes]
+      updatedNotes[updatedNoteIndex] = updatedNote
+      setNotes(updatedNotes)
+      setSelectedNote(null)
+    })
+    .catch((error) => console.error('Error:', error))
+  }
+
+  function handleNoteChange(event) {
+    const newNote = event.target.value
+    setSelectedNote((prevNote) => ({ ...prevNote, note: newNote }))
+  }
+
+  function handleSaveNote() {
+    if (selectedNote) {
+      updateNote(selectedNote.id, selectedNote.note)
+    }
+    setSelectedNote(null)
+    setIsEditing(false)
+  }
+
+  function handleCancelNote() {
+    setSelectedNote(null)
+    setIsEditing(false)
+  }
+
+  function handleEditNote(note) {
+    setSelectedNote(note)
+    setIsEditing(true)
+  }
+
+  function handleDeleteNote(note) {
+    setNoteToDelete(note)
+    setDeleteModalOpen(true)
+  }
+
+  function handleConfirmDelete() {
+    if (noteToDelete) {
+      deleteNote(noteToDelete.id)
+    }
+    setDeleteModalOpen(false)
+    setNoteToDelete(null)
+  }
+
+  function handleCancelDelete() {
+    setDeleteModalOpen(false)
+    setNoteToDelete(null)
+  }
+
+  function formatDateTime(dateTime) {
+    const date = new Date(dateTime)
+    const day = date.getDate()
+    const month = date.getMonth() + 1
+    const year = date.getFullYear()
+    const hours = date.getHours()
+    const minutes = date.getMinutes()
+    const seconds = date.getSeconds()
+
+    const formattedDay = day < 10 ? `0${day}` : day
+    const formattedMonth = month < 10 ? `0${month}` : month
+    const formattedHours = hours < 10 ? `0${hours}` : hours
+    const formattedMinutes = minutes < 10 ? `0${minutes}` : minutes
+    const formattedSeconds = seconds < 10 ? `0${seconds}` : seconds
+
+    return `${formattedDay}/${formattedMonth}/${year} 
```

Comment:
let's extract this into a utils function for formatting timestamps. Maybe there is something already existing in javascript